divert/core.py

```diff
@@ -16,6 +18,19 @@ def raise_again(self, *_: Any) -> bool:
         self._jumps_remaining -= 1
         return self._jumps_remaining > 0
 
+    @classmethod
+    def make_weak(cls, *args: Any, **kwargs: Any) -> Diversion:
+        """Make weak variant of the diversion class.
+
+        By dynamically forming it provides support for generic extensions of the base diversions.
+        The minor performance hit is acceptable vs. having to maintain `WeakDiversion` instances for each extension.
+        """
+
+        class WeakDiversion(cls, Exception):
```

Comment:
creating it inline also means that the classname is a little goofy. You can fix that by assigning to _both_ `__name__` and `__qualname__` of the class